sklearn/neural_network/tests/test_elm.py

```diff
@@ -0,0 +1,393 @@
+"""
+Testing for Extreme Learning Machines module (sklearn.neural_network)
+"""
+
+# Author: Issam H. Laradji
+# Licence: BSD 3 clause
+
+import sys
+
+import numpy as np
+
+from numpy.testing import assert_almost_equal, assert_array_equal
+from scipy.sparse import csr_matrix
+from itertools import product
+
+from sklearn import cross_validation
+from sklearn.datasets import load_digits, load_boston
+from sklearn.datasets import make_regression, make_multilabel_classification
+from sklearn.ensemble import AdaBoostClassifier
+from sklearn.externals.six.moves import cStringIO as StringIO
+from sklearn.metrics import roc_auc_score
+from sklearn.neural_network import ELMClassifier
+from sklearn.neural_network import ELMRegressor
+from sklearn.preprocessing import StandardScaler, MinMaxScaler
+from sklearn.utils import gen_batches
+from sklearn.utils.testing import assert_raises, assert_greater, assert_equal
+from sklearn.utils.testing import assert_not_equal
+
+
+np.seterr(all='warn')
+
+random_state = 1
+
+ACTIVATION_TYPES = ["logistic", "tanh", "relu"]
+CLASSIFICATION_TYPES = ["binary", "multi-class"]
+
+digits_dataset_multi = load_digits(n_class=3)
+
+Xdigits_multi = MinMaxScaler().fit_transform(digits_dataset_multi.data[:200])
+ydigits_multi = digits_dataset_multi.target[:200]
+
+digits_dataset_binary = load_digits(n_class=2)
+
+Xdigits_binary = MinMaxScaler().fit_transform(digits_dataset_binary.data[:200])
+ydigits_binary = digits_dataset_binary.target[:200]
+
+
+classification_datasets = {'binary': (Xdigits_binary, ydigits_binary),
+                           'multi-class': (Xdigits_multi, ydigits_multi)}
+
+boston = load_boston()
+
+Xboston = StandardScaler().fit_transform(boston.data)[:200]
+yboston = boston.target[:200]
+
+
+def check_elm(elm, type_, dataset, n_samples, score):
+    if type_ == 'classification':
+        X, y = classification_datasets[dataset]
+    elif type_ == 'regression':
+        X, y = Xboston, yboston
+
+    X_train, y_train = X[:n_samples], y[:n_samples]
+    X_test = X[n_samples:]
+
+    expected_shape = X_test.shape[0]
+    expected_dtype = y_train.dtype.kind
+
+    for activation in ACTIVATION_TYPES:
+        elm.fit(X_train, y_train)
+
+        y_predict = elm.predict(X_test)
+        assert_greater(elm.score(X_train, y_train), score)
+
+        assert_equal(y_predict.shape[0], expected_shape)
+        assert_equal(y_predict.dtype.kind, expected_dtype)
+
+
+def test_classification():
+    """Test ELMClassifier.
```

Comment:
All tests should use comments, not docstrings now, so we can more easily track what is happening.